docs/connectors/mongodb/native-operations/security-best-practices.mdx

```diff
@@ -0,0 +1,185 @@
+---
+sidebar_position: 5
+sidebar_label: Security Best Practices
+description:
+  "When writing native queries it is critical to understand the security implications of how arguments are interpreted."
+keywords:
+  - native queries
+  - native mutations
+  - security
+seoFrontMatterUpdated: false
+---
+
+# Security Best Practices in MongoDB Native Operations
+
+Native queries and native mutations allow you to construct any arbitrary query or database command respectively. That
+power comes with the potential to introduce exploits in your API which you should take care to avoid. In particular
+input parameters in native queries and native mutations are **not** sanitized so you must take care to escape uses of
+inputs. These are inputs that appear in your query or command using the double-curly-brace placeholder syntax: `{{
+parameter_name }}`.
+
+## Do not substitute ASTs
+
+Double-curly-brace parameter substitution technically makes it possible to accept pipeline expressions or stages (a.k.a
+"ASTs") as arguments. **Do not do this**.
+
+```json
+[
+  {
+    "$match": {
+      "user_id": { "$eq": "{{ user_id }}" }
+    }
+  },
+  "{{ custom_query_steps }}",
+  {
+    "$project": {
+      "title": true,
+      "content": true
+    }
+  }
+]
+```
+
+The use of `custom_query_steps` gives complete control to users to insert any code into the query. Even if there is no
+sensitive data in a collection, there are aggregation stages, like `$out`, that can write to the database, and therefore
+can **inject data**. There are also stages, like `$lookup`, that can **read data from other collections**.
+
+Contrary to stages, the author of this guide was not able to identify any expression operators that are capable of
+writing to the database, or of reading data from another collection. But the author is not 100% certain this is not
+possible.
```

Comment:
This makes us sound like we don't know what we're talking about. Can we backchannel with Mongo or something and get a concrete answer to this question?

Reply:
@hallettj — I totally agree with @daniel-chambers.

Additionally, the hierarchy of these documents is changing. We have #803 underway, which moves the Mongo (and other connectors') configuration documents (including native operations) into a `reference/connectors/[connector-name]` directory.

Can you please:
- Change your base branch to `rob/docs/add-data-sources-section`
- Relocate your new files + changes to the appropriate place
- Make modifications based on Daniel's feedback (if you need some help with wordsmithing, let me know)

Happy to answer any questions! 🤙

Reply:
I removed the paragraph in question. It's not really necessary for making decisions about how to secure a query.

@robertjdominguez I changed the PR base as you requested

Reply:
Thank you, @hallettj!